UPGRADE.md

```diff
@@ -0,0 +1,90 @@
+Note about upgrading: Doctrine uses static and runtime mechanisms to raise
+awareness about deprecated code.
+
+- Use of `@deprecated` docblock that is detected by IDEs (like PHPStorm) or
+  Static Analysis tools (like Psalm, phpstan)
+- Use of our low-overhead runtime deprecation API, details:
+  https://github.com/doctrine/deprecations/
+
+# Upgrade to 3.0
+
+## BC Break: removed `ObjectManager::merge()`
+
+`ObjectManagerDecorator::merge()` is removed as well.
```

Comment:
```suggestion
`ObjectManagerDecorator::merge()` is removed without replacement.
```

I believe "as well" part originates from times we wanted to remove detach, but since we're not let's rephrase